lib/api/api.go

```diff
@@ -1280,27 +1282,43 @@ func (s *service) getDBIgnores(w http.ResponseWriter, r *http.Request) {
 func (s *service) postDBIgnores(w http.ResponseWriter, r *http.Request) {
 	qs := r.URL.Query()
 
-	bs, err := ioutil.ReadAll(r.Body)
-	r.Body.Close()
+	ignores, err := deserializeIgnores(r.Body)
 	if err != nil {
 		http.Error(w, err.Error(), 500)
 		return
 	}
 
-	var data map[string][]string
-	err = json.Unmarshal(bs, &data)
+	err = s.model.SetIgnores(qs.Get("folder"), ignores)
 	if err != nil {
 		http.Error(w, err.Error(), 500)
 		return
 	}
 
-	err = s.model.SetIgnores(qs.Get("folder"), data["ignore"])
-	if err != nil {
+	s.getDBIgnores(w, r)
+}
+
+func (s *service) readIgnores(w http.ResponseWriter, r *http.Request) {
+	ignores, err := config.IgnoresFromFile(r.URL.Query().Get("path"))
```

Comment:
This is a generic file reader (readIgnores("/etc/passwd")), not sure if that matters or not as it's anyway a thing Syncthing can read and hence sync if asked to, but we should consider.

Reply:
Should we rename the method to something like "readfile" to make it explicit? I don't think there's a sensible way around having such an endpoint and still sanely handle ignores on new folders (i.e. not clobber any existing .stignore files).

Reply:
We could maybe have a `clobber: false` attribute on the set-ignores endpoint, which only sets the ignores if the file is not already present... And then use that if the user creates a folder with the default unmodified ignores...

Reply:
If we want to get rid of this, I'd propose delaying setting ignore patterns to after the folder is created. That already happens behind the scenes by first adding the folder paused, then applying ignores, then unpausing - I'd just reflect that in the UI too. So when adding a folder instead of an input field, there would just a be a short explanation and a checkbox. If ticked, one can edit the ignores after clicking save (or add or whatever the confirmation button is).

Reply:
That checkbox would leave the folder paused, you mean? And at that point we go through the normal read-ignores and/or use the default? The user then unpauses the folder when done. I guess that would work.

Having an attribute to set-ignores to say "only write if there aren't already any ignores" would also work though I think.

Reply:
I meant that, but automatically: User clicks save, all input elements get frozen and the ignores checkbox switched to an ignores input populated through the usual "give me this folders ignore patterns"  endpoint. Then they click save again and the folder gets unpaused. Lets do that actually. 

And that kind of flag seems like a mixed bag UX wise: I tell syncthing to set some ignores, and then it doesn't actually do that to not clobber existing ones.